### DISCUSSION TITLE
问题咨询

### DISCUSSION THREAD

**Discussion by GitChenfan** [wind-c/comqtt#82] [General]
你好。
想咨询下，
通过 auth 后将一部分用户信息进行存储，保存在当前连接的session下，需要怎么做，其他hook也可以获取到该session
比如 连接的设备基本信息，用户信息，权限等。

**Comment by wind-c** [comment]
不知道你设计上为何要保存这些信息，如果非得如此，最好是拓展clients.go#Client struct，加保存这些信息的字段。

**Reply by GitChenfan** [reply]
主要是业务上需要对连接的设备上报及订阅的主题做一些权限判断。

**Reply by GitChenfan** [reply]
因为只想通过添加插件的方式完成业务内容，方便后续更新
